### PR TITLE
feat: Implement display articles with translation for the activity stream - EXO-73599 - Meeds-io/MIPs#128

### PR DESCRIPTION
**Pull request by sofyenne**

### CODE REVIEW

File: content-service/src/main/java/io/meeds/news/service/NewsService.java [hunk 1]
```diff
@@ -159,6 +159,15 @@ News getNewsByIdAndLang(String newsId,
    */
   News getNewsArticleById(String newsId);
 
+  /**
+   * Retrives a news identified by its technical identifier
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/NewsService.java]:
```suggestion
   * Retrieves a news identified by its technical identifier and lang
```

File: content-service/src/main/java/io/meeds/news/service/NewsService.java [hunk 2]
```diff
@@ -159,6 +159,15 @@ News getNewsByIdAndLang(String newsId,
...
+   *
+   * @param newsId {@link News} identifier
+   * @param lang news translation language
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/NewsService.java]:
```suggestion
   * @param lang {@link News} translation language
```

File: content-service/src/main/java/io/meeds/news/service/NewsService.java [hunk 3]
```diff
@@ -226,6 +235,23 @@ News getNewsByActivityId(String activityId,
                            org.exoplatform.services.security.Identity currentIdentity) throws IllegalAccessException,
                                                                                        ObjectNotFoundException;
 
+  /**
+   * Retrieves a news item identified by originating Activity identifier or a
+   * shared activity identifier
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/NewsService.java]:
```suggestion
   * Retrieves a {@link News} by its related activity identifier or its shared activity identifier
```

File: content-service/src/main/java/io/meeds/news/service/NewsService.java [hunk 4]
```diff
@@ -226,6 +235,23 @@ News getNewsByActivityId(String activityId,
...
+   *
+   * @param activityId {@link ExoSocialActivity} identifier
+   * @param currentIdentity user attempting to access news
+   * @param lang news translation version
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/NewsService.java]:
```suggestion
   * @param lang {@link News} translation language
```

File: content-service/src/main/java/io/meeds/news/service/NewsService.java [hunk 5]
```diff
@@ -226,6 +235,23 @@ News getNewsByActivityId(String activityId,
...
+   * @return {@link News} if found else null
+   * @throws IllegalAccessException when user doesn't have access to
+   *           {@link News} or {@link ExoSocialActivity}
+   * @throws ObjectNotFoundException when a {@link News} wasn't found for this
+   *           activity identifier
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/NewsService.java]:
```suggestion
   * @throws ObjectNotFoundException when a {@link News} wasn't found for the given
   *           activity identifier
```

File: content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java [hunk 6]
```diff
@@ -93,6 +97,22 @@ public void processActivity(ExoSocialActivity activity) {
...
+          articleLanguages.stream().forEach(articleLanguage -> {
+            News articleTranslation = newsService.getNewsArticleByIdAndLang(newsId, articleLanguage);
+            if (articleTranslation != null) {
+              newsTranslations.put("news_" + articleLanguage, articleTranslation);
+            }
+          });
+          activity.getLinkedProcessedEntities().put("newsTranslations", newsTranslations);
+        }
+      } catch (Exception exception) {
+        LOG.error("error when adding the news translation to the activity linked processed entities", exception);
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java]:
```suggestion
        LOG.error("Error when adding the news translation to the activity linked processed entities", exception);
```

File: content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java [hunk 7]
```diff
@@ -93,6 +97,22 @@ public void processActivity(ExoSocialActivity activity) {
         LOG.warn("Error retrieving news with id {}", activity.getTemplateParams().get("newsId"), e);
       }
       activity.getLinkedProcessedEntities().put("news", news);
+      try {
+        String newsId = news.getId();
+        List<String> articleLanguages = newsService.getArticleLanguages(newsId, false);
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java:102]: In the implem of this method .getArticleLanguages, withDrafts is always true, to be verified

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java [hunk 8]
```diff
@@ -2038,6 +2056,7 @@ private News addNewArticleVersionWithLang(News news, Identity versionCreator) th
       if (draftPage != null) {
         deleteDraftArticle(draftPage.getId(), draftPage.getAuthor());
       }
+      NewsUtils.broadcastEvent(NewsUtils.ADD_ARTICLE_TRANSLATION, versionCreator, news);
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java:2037]: to be discussed

File: content-webapp/src/main/webapp/vue-app/news-details/components/ExoNewsDetailsActivity.vue [hunk 9]
```diff
@@ -101,16 +101,33 @@ export default {
...
+            this.selectedTranslation = this.originalVersion;
+          }
         }
       } else {
-        this.$newsServices.getNewsByActivityId(this.activityId)
+        this.$newsServices.getNewsByActivityId(this.activityId, this.selectedTranslation.value)
           .then(news => {
-            if (news?.id) {
-              this.getArticleVersionWithLang(news.id, this.selectedTranslation.value);
+            if (this.previousSelectedTranslation === 'autoTranslation') {
```
Comment (azayati) [inline content-webapp/src/main/webapp/vue-app/news-details/components/ExoNewsDetailsActivity.vue:121]: autoTranslation is not available for meeds, to be discussed

**Review by azayati — CHANGES_REQUESTED** [review]

File: content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java (hunk 7 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java:102]: https://github.com/Meeds-io/content/blob/feature/experience/content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java#L1138

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java (hunk 8 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java:2037]: This is to refresh the attached activity cache after adding or deleting an article translation!

File: content-webapp/src/main/webapp/vue-app/news-details/components/ExoNewsDetailsActivity.vue (hunk 9 above)
Comment (sofyenne) [reply content-webapp/src/main/webapp/vue-app/news-details/components/ExoNewsDetailsActivity.vue:121]: So, it is a condition only for eXo!

Comment (azayati) [reply content-webapp/src/main/webapp/vue-app/news-details/components/ExoNewsDetailsActivity.vue:121]: I mean that we should not mention "autoTranslation" addon into meeds codebase

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java [hunk 10]
```diff
@@ -196,6 +197,9 @@ public class NewsServiceImpl implements NewsService {
 
   @Autowired
   private NewsSearchConnector newsSearchConnector;
+    @Qualifier("newsService")
+    @Autowired
+    private NewsService newsService;
```
Comment (azayati) [inline content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java]: indentation

**Review by azayati — CHANGES_REQUESTED** [review]

File: content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java (hunk 7 above)
Comment (azayati) [reply content-service/src/main/java/io/meeds/news/activity/processor/ActivityNewsProcessor.java:102]: update the description of apiResponses for the rest service getAvailableTranslationLanguages

**Review by azayati — CHANGES_REQUESTED** [review]

File: content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java (hunk 10 above)
Comment (sofyenne) [reply content-service/src/main/java/io/meeds/news/service/impl/NewsServiceImpl.java]: unnecessary injection of NewsService removed !

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**